Add support for multiple attr for select input type

RubenMartins opened the pull request:
Props to @rubengc #995 

Didn't found this in a PR format so created this one.

Please advise.

<!--- Provide a general summary of your changes in the Title above -->

## Description
Add support for attribute multiple, showing correctly the multiple elements saved on database

## Motivation and Context
Motivation to provide CMB2 with the multiple feature out of box easily.
Fixes #995 

## Risk Level
admin-only

## Testing procedure
You can test by adding the 'multiple' attribute to a select field type
`$cmb->add_field( array( 'name' => 'Test Select', 'id' => 'wiki_test_select', 'type' => 'select', 'options' => array( 'standard' => __( 'Option One', 'cmb2' ), 'custom' => __( 'Option Two', 'cmb2' ), 'none' => __( 'Option Three', 'cmb2' ), ), 'attributes' => array( 'multiple' => true ) ) );`

## Types of changes
**Bug fix (non-breaking change which fixes an issue)**

## Checklist:
<!--- Go over all the following points, and put an x in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My code and pull requests meets the [Contributing guidelines](https://github.com/CMB2/CMB2/blob/develop/CONTRIBUTING.md).

RubenMartins committed:
Add support for multiple attr for select input type

Props to @rubengc #995 

Didn't found this in a PR format so created this one.

Please advise.

jtsternberg reviewed (changes requested):
1. There are some code-formatting issues. 
2. `CMB2_Utils::normalize_if_numeric( $opt_value )` should be used where you're using just `$opt_value`.
3. `in_array` should have the strict mode set, by setting 3rd param to `true`
4. Last else-if is a repeat of the first if?
5. Provide testing instructions please.

RubenMartins committed:
Improve code

This can be tested against a select field with attr "multiple":
$cmb->add_field( array(
	'name'             => 'Test Select',
	'id'               => 'wiki_test_select',
	'type'             => 'select',
	'options'          => array(
		'standard' => __( 'Option One', 'cmb2' ),
		'custom'   => __( 'Option Two', 'cmb2' ),
		'none'     => __( 'Option Three', 'cmb2' ),
	),
	'attributes' => array(
		'multiple' => true
	),
) );

RubenMartins commented:
@jtsternberg This can be tested against a select field with the attribute "multiple":
`$cmb->add_field( array(
	'name'             => 'Test Select',
	'id'               => 'wiki_test_select',
	'type'             => 'select',
	'options'          => array(
		'standard' => __( 'Option One', 'cmb2' ),
		'custom'   => __( 'Option Two', 'cmb2' ),
		'none'     => __( 'Option Three', 'cmb2' ),
	),
	'attributes' => array(
		'multiple' => true
	)
) );`

By using the attribute multiple this will submit multiple values in the same meta. This code recognizes that the value is an array and acts accordingly to save the data correctly.

This is what I remember I did to fix the issue that I faced with a select with multiple values.
